pyccel/ast/utilities.py

```diff
@@ -76,32 +70,44 @@ def builtin_function(expr, args=None):
 
     return None
 
+#==============================================================================
+decorators_mod = Module('decorators',(),
+        funcs = [PyccelFunctionDef(d, PyccelInternalFunction) for d in pyccel_decorators.__all__])
+pyccel_mod = Module('pyccel',(),(),
+        imports = [Import('decorators', decorators_mod)])
 
 # TODO add documentation
-builtin_import_registery = {'numpy': {
-                                      **numpy_functions,
-                                      **numpy_constants,
-                                      'linalg':numpy_linalg_functions,
-                                      'random':numpy_random_functions
-                                      },
-                            'numpy.linalg': numpy_linalg_functions,
-                            'numpy.random': numpy_random_functions,
-                            'scipy.constants': scipy_constants,
-                            'itertools': {'product': Product},
-                            'math': {**math_functions, ** math_constants},
-                            'pyccel.decorators': pyccel_decorators.__all__}
+builtin_import_registery = Module('__main__',
+        (),(),
+        imports = [
+            Import('numpy', AsName(numpy_mod,'numpy')),
+            Import('scipy', AsName(scipy_mod,'scipy')),
+            Import('itertools', AsName(itertools_mod,'itertools')),
+            Import('math', AsName(math_mod,'math')),
+            Import('pyccel', AsName(pyccel_mod,'pyccel'))
+            ])
 if sys.version_info < (3, 10):
     from .builtin_imports import python_builtin_libs
 else:
     python_builtin_libs = set(sys.stdlib_module_names) # pylint: disable=no-member
 
 recognised_libs = python_builtin_libs.union(builtin_import_registery.keys())
```

Comment:
It has nothing to do with this PR, but... using the pipe operator `|` for the union is more Pythonic:
```suggestion
recognised_libs = python_builtin_libs | builtin_import_registery.keys()
```

Reply:
Huh, I have never seen that before